packages/code-studio/src/dashboard/plugins/ConsolePlugin.tsx

```diff
@@ -0,0 +1,200 @@
+      if (
+        // Look for stack with matching component type, don't allow empty and don't create new
+        !LayoutUtils.getStackForComponentTypes(
+          layout.root,
+          [component],
+          false,
+          true,
+          false
+        )
+      ) {
+        itemsAdded = true;
+        log.debug(`${component} not found - creating new`);
+        const config = {
+          component,
+          isClosable: false,
+          props: { metadata: {} },
+          title,
+          type: 'react-component',
+        };
+        stack.addChild(config);
+      }
+    });
+
+    if (itemsAdded) {
+      // Last added item becomes active, restore activeItem
+      // Fall back to first item in stack
+      stack.setActiveContentItem(
+        activeItem != null ? activeItem : stack.contentItems[0]
+      );
+    }
+
+    const logStack = getStackForComponentType(LogPanel.COMPONENT);
+
+    if (!logStack) {
+      log.debug('LogPanel not found - creating new');
+      const config = {
+        component: LogPanel.COMPONENT,
+        isClosable: false,
+        title: LogPanel.TITLE,
+        type: 'react-component',
+        props: { session: null },
+        id: shortid.generate(),
+      };
+      consoleStack.addChild(config);
+    }
+  }, [getConsoleStack, getStackForComponentType, layout]);
+
+  const handleSendCommand = useCallback(
+    (command = string, focus = true, execute = true) => {
+      const trimmedCommand = command && command.trim();
+      if (!trimmedCommand) {
+        log.info('Ignoring empty code');
+      } else {
+        const consolePanel = getConsolePanel();
+        if (!consolePanel) {
+          log.error('Console panel not found');
+          return;
+        }
+        log.debug('Send command: ', command, focus, execute);
+        consolePanel.addCommand(command, focus, execute);
+      }
+    },
+    [getConsolePanel]
+  );
+
+  const handleSettingsChanged = useCallback(() => {
+    // TODO: needs to store data with the dashboard itself
+    log.warn('handleSettingsChanged not implemented yet!!');
+  }, []);
+
+  useEffect(() => {
+    const cleanups = [
+      registerComponent(
+        ConsolePanel.COMPONENT,
+        (ConsolePanel as unknown) as ComponentType
+      ),
+      registerComponent(
+        CommandHistoryPanel.COMPONENT,
+        (CommandHistoryPanel as unknown) as ComponentType
+      ),
+      registerComponent(
+        FileExplorerPanel.COMPONENT,
+        (FileExplorerPanel as unknown) as ComponentType
+      ),
+      registerComponent(
+        LogPanel.COMPONENT,
+        (LogPanel as unknown) as ComponentType
+      ),
+      registerComponent(
+        NotebookPanel.COMPONENT,
+        (NotebookPanel as unknown) as ComponentType
+      ),
+    ];
+
+    return () => {
+      cleanups.forEach(cleanup => cleanup());
+    };
+  }, [registerComponent]);
+
+  useEffect(() => {
+    addMissingPanels();
```

Comment:
Not only does it not seem to work... but this might be functionality we should only have on Enterprise side? With application mode in community, we allow exporting of layouts, and re-importing without ConsolePanel or whatever panels you don't want. By adding them back, we're not doing what the user may expect.

Reply:
I'm just going to remove this community side.